File: MachineLearning/perceptron/perceptron.js
class Perceptron {
  weights = [0, 0];
  constructor() {
    // Inicializando com valores random
    for (let i = 0; i < this.weights.length; i++) {
      this.weights[i] = random(-1, 1);
    }
  }

  //inputs --> entrada array de duas posiçoes (X1 e X2)
  guess(inputs) {
    let sum = 0;
    // aplicando a formula (Y = X1 * W1 + X2 * W2)
    for (let i = 0; i < this.weights.length; i++) {
      sum += inputs[i] * this.weights[i];
    }
    const output = sign(sum);
    return output;
  }
}

//Função de aticação
function sign(num) {
  return num >= 0 ? 1 : -1;
}

File: Perceptron-Python/perceptron.py
import random

class Perceptron:
    def __init__(self, num_weights):
        self.weights = [random.uniform(-1, 1) for _ in range(num_weights)]
        self.lr = 0.01 

    def guess(self, inputs):
        total = sum(inputs[i] * self.weights[i] for i in range(len(self.weights)))
        return self._sign(total)

    def train(self, inputs, target):
        guess = self.guess(inputs)
        error = target - guess
        for i in range(len(self.weights)):
            self.weights[i] += error * inputs[i] * self.lr

    def _sign(self, num):
        return 1 if num >= 0 else -1